From: Akhil Mohan <akhil.mohan@oracle.com>
Description: Adding extra search path for testsuite.

--- a/myblockchain-test/lib/mtr_cases.pm	2014-02-24 13:14:37 +0530
+++ b/myblockchain-test/lib/mtr_cases.pm	2014-07-02 11:46:24 +0530
@@ -288,6 +288,7 @@
     {
       $suitedir= my_find_dir($::basedir,
 			     ["share/myblockchain-test/suite",
+			      "lib/myblockchain-test/suite",
 			      "myblockchain-test/suite",
 			      "internal/myblockchain-test/suite",
 			      "myblockchain-test",
